axelrod/tests/unit/test_tournament.py

```diff
@@ -177,44 +304,74 @@ def test_progress_bar_play(self):
             turns=axelrod.DEFAULT_TURNS,
             repetitions=self.test_repetitions)
 
-        results = tournament.play()
+        err = io.StringIO()
+        with redirect_stderr(err):
+            results = tournament.play()
         self.assertIsInstance(results, axelrod.ResultSet)
-        self.assertEqual(tournament.progress_bar.total, 15)
```

Comment:
i found a solution with `patch` and a new class I made, `RecordedTQDM(tqdm)`.  

While testing it out, i discovered that if you play the same tournament repeatedly, `num_interactions` does not reset with each play.  this affects the `"Analysing: "` progress_bar.  For instance, if `play` creates 75 interactions, the second time you `play` there are 150 interactions.  The "Analysing: " progress bar is now `total=150` but finishes at `n=75`. Should `num_interactions` be resetting with each play?